cluster/src/dunit/scala/io/snappydata/cluster/DDLRoutingDUnitTest.scala

```diff
@@ -194,30 +214,48 @@ class DDLRoutingDUnitTest(val s: String) extends ClusterManagerTestBase(s) {
           throw sqle
         }
     }
-    // Commented due to SNAP-1818
-//    s.execute(s"ALTER TABLE $tableName SET GATEWAYSENDER ()")
-//    s.execute("CREATE ASYNCEVENTLISTENER myListener (" +
-//      " listenerclass 'com.pivotal.gemfirexd.callbacks.DBSynchronizer'" +
-//      " initparams 'org.apache.derby.jdbc.EmbeddedDriver,jdbc:derby:newDB;create=true')")
-//
-//    s.execute(s"ALTER TABLE $tableName SET ASYNCEVENTLISTENER (myListener) ")
-//    var rs = s.executeQuery(s"select * from SYS.SYSTABLES where tablename='$tableName'")
-//    while (rs.next) {
-//      assert("MYLISTENER".equalsIgnoreCase(rs.getString(17)))
-//    }
-//    s.execute(s"ALTER TABLE $tableName SET ASYNCEVENTLISTENER () ")
-//    rs = s.executeQuery("select * from SYS.SYSTABLES where tablename='TESTTABLE'")
-//    while (rs.next) {
-//      assert(rs.getString(17) == null)
-//    }
+
+    // asynceventlistener
+    s.execute("CREATE ASYNCEVENTLISTENER myListener (" +
+      " listenerclass 'com.pivotal.gemfirexd.callbacks.DBSynchronizer'" +
+      " initparams 'org.apache.derby.jdbc.EmbeddedDriver,jdbc:derby:newDB;create=true')" +
+      " server groups(sg1)")
+
+    s.execute(s"ALTER TABLE $tableName SET ASYNCEVENTLISTENER (myListener) ")
+    var rs = s.executeQuery(s"select * from SYS.SYSTABLES where tablename='$tableName'")
+    while (rs.next) {
+      assert("MYLISTENER".equalsIgnoreCase(rs.getString(17)))
+    }
+    s.execute(s"ALTER TABLE $tableName SET ASYNCEVENTLISTENER () ")
+    rs = s.executeQuery(s"select * from SYS.SYSTABLES where tablename='$tableName'")
+    while (rs.next) {
+      assert(rs.getString(17) == null)
+    }
+    s.execute(s"drop ASYNCEVENTLISTENER myListener")
+
+    // gatewaysenders/receivers
+    s.execute("CREATE GATEWAYSENDER gwSender ( REMOTEDSID 2) SERVER GROUPS (sg1)")
+    s.execute("CREATE GATEWAYRECEIVER gwRcvr (startport 1111 endport 9999) SERVER GROUPS (sg1)")
+    s.execute(s"ALTER TABLE $tableName SET GATEWAYSENDER (gwSender) ")
+    rs = s.executeQuery(s"select * from SYS.SYSTABLES where tablename='$tableName'")
+    while (rs.next) {
+      assert("gwSender".equalsIgnoreCase(rs.getString(19)))
+    }
+    s.execute(s"ALTER TABLE $tableName SET GATEWAYSENDER () ")
+    rs = s.executeQuery(s"select * from SYS.SYSTABLES where tablename='$tableName'")
+    while (rs.next) {
+      assert(rs.getString(19) == null)
+    }
+    s.execute(s"drop GATEWAYSENDER gwSender")
```

Comment:
Hey I think it will be nice to have a restart of the system and see if after ddl replay all the listeners are there as per expectations. First test with async on snappydata cluster.